src/System/extract_state.rs

```diff
@@ -101,8 +106,9 @@ fn get_location_id(locations: &mut Iter<&str>, machine: &SystemRecipe) -> Locati
             Box::new(get_location_id(locations, left)),
             Box::new(get_location_id(locations, right)),
         ),
-        SystemRecipe::Component(_comp) => {
-            LocationID::Simple(locations.next().unwrap().trim().to_string())
-        }
+        SystemRecipe::Component(..) => match locations.next().unwrap().trim() {
```

Comment:
Consider handling the two cases from the option instead of unwrapping

Reply:
We have already handled this problem in `validate_reachability_input `defined in `extract_system_rep`.